src/serialization/data_file.cpp

```diff
@@ -0,0 +1,15 @@
+#include "serialization/data_file.hpp"
+#include <stdexcept>
+
+namespace found {
+
+    void DataFileHeader::validate() {
+        if (magic[0] != 'F' || magic[1] != 'O' || magic[2] != 'U' || magic[3] != 'N') {
```

Comment:
I think I know why you've chosen this magic string, but you might want to explain why this is (via documentation).